File: Class_Member/2.py
class Cs:
    @staticmethod
    def static_method():
        print('Static method')
    @classmethod
    def class_method(cls): # cls를 첫번째 매개변수로 갖는다.
        print('Class method')
    def instance_method(self):
        print('Instance_method')

i = Cs()
Cs.static_method()
Cs.class_method()
i.instance_method()

# 클래스 소속: staticmethod, classmethod
# vs
# 인스턴스 소속: instancemethod

File: Class_Member/2.rb
class Cs
  # 'Cs.'를 꼭 붙여줘야 한다!
  def Cs.class_method() # 클래스에 소속된 멤버라는 것을 명시!
    p "Class method"
  end
  def instance_method()
    p "instance_method"
  end
end

i = Cs.new()
Cs.class_method() # 클래스 소속
i.instance_method() #인스턴스 소속
# Error <= i.class_method() # 클래스 소속
# Error <= Cs.instance_method() #인스턴스 소속

# 클래스 메소드는 클래스의 소속으로 사용한다.
# 인스턴스 메소드는 인스턴스의 소속으로 사용한다.
